File: clinic-api/src/main/java/com/clinic/entity/DentalImage.java
package com.clinic.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("dental_image")
public class DentalImage implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long clinicId;
    private Long patientId;
    private Long medicalRecordId;
    private String imageType;
    private String imageUrl;
    private String thumbnailUrl;
    private String fileName;
    private Long fileSize;
    private String mimeType;
    private Integer width;
    private Integer height;
    private LocalDate shotDate;
    private String bodyPart;
    private String description;
    private Integer sortOrder;

    @TableField(exist = false)
    private List<String> linkedTeeth;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}

File: clinic-admin/src/api/dentalImage.ts
import { get, post } from '@/utils/request'

export interface DentalImage {
  id?: number
  clinicId: number
  patientId: number
  medicalRecordId?: number
  imageType: string
  imageUrl: string
  thumbnailUrl?: string
  fileName?: string
  fileSize?: number
  mimeType?: string
  width?: number
  height?: number
  shotDate?: string
  bodyPart?: string
  description?: string
  sortOrder?: number
  linkedTeeth?: string[]
}

export const getDentalImageList = (params: {
  clinicId: number
  patientId: number
  imageType?: string
  pageNum?: number
  pageSize?: number
}) => get<{ total: number; list: DentalImage[] }>('/admin/dental-image/list', params)

export const getDentalImageDetail = (id: number) =>
  get<DentalImage>(`/admin/dental-image/${id}`)

export const createDentalImage = (data: Partial<DentalImage>) =>
  post<DentalImage>('/admin/dental-image/create', data)

export const updateDentalImage = (data: Partial<DentalImage>) =>
  post('/admin/dental-image/update', data)

export const deleteDentalImage = (id: number, clinicId: number) =>
  post('/admin/dental-image/delete', { id, clinicId })

export const linkTeethToImage = (data: { imageId: number; toothNumbers: string[] }) =>
  post('/admin/dental-image/link-teeth', data)

export const getImagesByRecord = (medicalRecordId: number) =>
  get<DentalImage[]>(`/admin/dental-image/by-record/${medicalRecordId}`)
